jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HostHeaderCustomizer.java

```diff
@@ -70,6 +71,17 @@ public Request customize(Request request, HttpFields.Mutable responseHeaders)
         builder.add(request.getHeaders());
         HttpFields headers = builder.asImmutable();
 
+        ConnectionMetaData connectionMetadata = new ConnectionMetaData.Wrapper(request.getConnectionMetaData())
+        {
+            private HostPort _serverAuthority = new HostPort(host, port);
+
+            @Override
+            public HostPort getServerAuthority()
+            {
+                return _serverAuthority;
+            }
+        };
```

Comment:
I don't think this is correct.
This customizer only needs to add the `Host` header, not change the server authority.